src/api/services/user-service.ts

```diff
@@ -0,0 +1,209 @@
+    if (!existingUser || existingUser.isDeleted === "Yes") {
+      throw new Error("No user found with userId");
+    } else {
+      return existingUser;
+    }
+  } catch (err) {
+    throw new Error("No user found with userId");
+  }
+};
+
+const login = async (email: string, password: string) => {
+  const tokenPassword = String(process.env.tokenPassword);
+  let existingUser;
+  try {
+    existingUser = await User.findOne({ email: email });
+  } catch (err) {
+    throw new Error("Login failed, Please try again");
+  }
+
+  if (!existingUser || existingUser.isDeleted === "Yes") {
+    throw new Error("User does not exist");
+  }
+
+  let isPasswordValid;
+  try {
+    isPasswordValid = await bcrypt.compare(password, existingUser.password);
+  } catch (err) {
+    throw new Error("Wrong credentials, Please try again");
+  }
+
+  if (!isPasswordValid) {
+    throw new Error("Wrong credentials, Please try again");
+  }
+
+  const tokenData = {
+    name: existingUser.firstName + existingUser.lastName,
+    email: existingUser.email,
+  };
+  let token;
+  try {
+    token = jwt.sign(tokenData, tokenPassword, { expiresIn: "1h" });
+  } catch (err) {
+    throw new Error("Token creation failed");
+  }
+
+  let decodedToken:JwtPayload;
+  try {
+    decodedToken = jwt.verify(token, tokenPassword) as JwtPayload;
+  } catch (err) {
+    throw new Error("Failed to verify token");
+  }
+  let userId = existingUser._id;
+
+  const newRefreshToken = new RefreshToken({
+    userId,
+    token,
+    created_at: decodedToken.iat,
+    expires_at: decodedToken.exp,
+  });
+
+  try {
+    await newRefreshToken.save();
+  } catch (err) {
+    throw new Error("Failed to save token");
+  }
+
+  const tokenResponse = {
+    ...tokenData,
+    token: token,
+  };
+
+  return tokenResponse;
+};
+
+const createUser = async (
+  email: string,
+  firstName: string,
+  lastName: string,
+  password: string
+) => {
+  const tokenPassword = process.env.tokenPassword;
+  let existingUser;
+  try {
+    existingUser = await User.findOne({ email: email });
+  } catch (err) {
+    throw new Error("Failed to create user");
+  }
+
+  if (existingUser) {
+    throw new Error("Email has alreay been used, Please try another email");
+  }
+
+  let hashPassword;
+  try {
+    hashPassword = await bcrypt.hash(password, 12);
+  } catch (err) {
+    throw new Error("Could not create user");
+  }
+
+  let isDeleted = "No";
+
+  const createUser = new User({
+    email,
+    firstName,
+    lastName,
+    password: hashPassword,
+    isDeleted,
+  });
+
+  try {
+    await createUser.save();
+  } catch (err) {
+    throw new Error("Failed to create user");
+  }
+
+  // const tokenData = {
+  //   name: createUser.firstName + createUser.lastName,
+  //   email: createUser.email,
+  // };
+  // let token;
+  // try {
+  //   token = jwt.sign(tokenData, tokenPassword, { expiresIn: "1h" });
+  // } catch (err) {
+  //   throw new Error("Token creation failed");
+  // }
```

Comment:
remove this if it is not required. Try not to check in comments or unused imports/variables

Reply:
YES SIR